app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt

```diff
@@ -68,16 +74,40 @@ import org.mozilla.fenix.wallpapers.Wallpaper
 @Composable
 @Suppress("LongParameterList")
 fun WallpaperSettings(
+    onLearnMoreClick: () -> Unit,
     wallpapers: List<Wallpaper>,
     defaultWallpaper: Wallpaper,
     loadWallpaperResource: suspend (Wallpaper) -> Bitmap?,
     selectedWallpaper: Wallpaper,
     onSelectWallpaper: (Wallpaper) -> Unit,
     onViewWallpaper: () -> Unit,
 ) {
-    val coroutineScope = rememberCoroutineScope()
-    val scaffoldState = rememberScaffoldState()
+    val wallpapersMap = remember(wallpapers) {
+        wallpapers
+            .groupBy { it.collection.name }
+    }
+    val mutableGroup = wallpapersMap.toMutableMap()
+    val defaultWallpapers = wallpapersMap["default"]?.toMutableList()
+    val classicWallpapers =
+        wallpapersMap["classic-firefox"]?.toMutableList() ?: emptyList<Wallpaper>()
 
+    defaultWallpapers?.addAll(classicWallpapers)
+    mutableGroup["default"] = defaultWallpapers ?: emptyList<Wallpaper>()
+    mutableGroup.remove("classic-firefox")
+
+    val groups = mutableGroup.map {
+        WallpaperProviderGroup(
+            wallpapers = it.value,
+            title = it.key,
+        )
+    }.sortedBy { it.title == "default" }
+
+    val maps = HashMap<WallpaperProviderGroup, List<Wallpaper>>()
+    val scaffoldState = rememberScaffoldState()
+    val coroutineScope = rememberCoroutineScope()
+    groups.forEach {
+        maps[it] = it.wallpapers
+    }
```

Comment:
I think we should move this grouping logic into an extension function located in a new file at `wallpapers/Extensions.kt` that has the signature `List<Wallpaper>.groupByCollection(): Map<Collection, List<Wallpaper>>`. We talked offline a bit, but this is an example suggestion that I think would get the behavior we desire here:

```
fun List<Wallpaper>.groupByCollection(): Map<Wallpaper.Collection, List<Wallpaper>> = groupBy {
    it.collection
}.filter {
    it.key.name != "default"
}.map {
    if (it.key.name == "classic-firefox") {
        it.key to listOf(Wallpaper.Default) + it.value
    } else {
        it.key to it.value
    }
}.toMap()
```